app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -1,7 +1,12 @@
-class Course::Discussion::PostsController < Course::ComponentController
```

Comment:
I agree with making it a concern. Concrete post controllers need to inherit form their own component controller, to get the essential information like the component instance, breadcrumbs, etc.